ui/app/AppLayouts/Communities/panels/AirdropsSettingsPanel.qml

```diff
@@ -43,11 +43,92 @@ StackView {
     signal registerAirdropFeeSubscriber(var feeSubscriber)
     signal enableNetwork(int chainId)
 
+    Loader {
+        id: assetsModelLoader
+        active: d.loadModels && root.communityTokens
+
+        sourceComponent: SortFilterProxyModel {
+
+            sourceModel: root.communityTokens
+            filters: ValueFilter {
+                roleName: "tokenType"
+                value: Constants.TokenType.ERC20
+            }
+            proxyRoles: [
+                ExpressionRole {
+                    name: "category"
+
+                    // Singleton cannot be used directly in the expression
+                    readonly property int category: TokenCategories.Category.Own
+                    expression: category
+                },
+                ExpressionRole {
+                    name: "iconSource"
+                    expression: model.image
+                },
+                ExpressionRole {
+                    name: "key"
+                    expression: model.symbol
+                },
+                ExpressionRole {
+                    name: "communityId"
+                    expression: ""
+                }
+            ]
+        }
+    }
+
+    Loader {
+        id: collectiblesModelLoader
+        active: d.loadModels && root.communityTokens
+
+        sourceComponent: SortFilterProxyModel {
+
+            sourceModel: root.communityTokens
+            filters: [
+                ValueFilter {
+                    roleName: "tokenType"
+                    value: Constants.TokenType.ERC721
+                },
+                ExpressionFilter {
+                    function getPrivileges(privilegesLevel) {
+                        return privilegesLevel === Constants.TokenPrivilegesLevel.Community ||
+                                (root.isOwner && privilegesLevel === Constants.TokenPrivilegesLevel.TMaster)
+                    }
+
+                    expression: { return getPrivileges(model.privilegesLevel) }
+                }
+            ]
+            proxyRoles: [
+                ExpressionRole {
+                    name: "category"
+
+                    // Singleton cannot be used directly in the epression
+                    readonly property int category: TokenCategories.Category.Own
+                    expression: category
+                },
+                ExpressionRole {
+                    name: "iconSource"
+                    expression: model.image
+                },
+                ExpressionRole {
+                    name: "key"
+                    expression: model.symbol
+                },
+                ExpressionRole {
+                    name: "communityId"
+                    expression: ""
+                }
+            ]
+        }
+    }
+
     function navigateBack() {
         pop(StackView.Immediate)
     }
 
     function selectToken(key, amount, type) {
+        d.loadModels = true
```

Comment:
I think you could make this a binding instead of imperative code; ie. load the models when the `StackView`'s current page is xxx